### PR TITLE
Provide high-level content about the Cobra static analyzer

### PR DESCRIPTION
**Pull request by mjeronimo**
Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

### CODE REVIEW

File: source/Related-Projects/Cobra.rst [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+Cobra can be used in one of three modes:
+
+* **Interactive** -- Cobra can be used interactively to load the token-based representation of a complete code base and query patterns in the code using a simple query language
+* **Program-based** -- Cobra can execute programs that can contain arbitrary branching and iteration over the token stream to identify more complex types of patterns
+* **Toolkit** -- Cobra can be used as an infrastructure for building more elaborate standalone checkers that are compiled separately and linked with the Cobra code that builds the central data structure
+
+How is Cobra used in Space ROS?
+-------------------------------
+
+The Space ROS Docker image incorporates the *cobra* command-line tool.
```
Comment (nuclearsandwich) [inline source/Related-Projects/Cobra.rst]:
```suggestion
The Space ROS Docker image incorporates the *cobra* command-line tool via the *cobra_vendor* package..
```

File: source/Related-Projects/Cobra.rst [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+In addition, there is a Python-based wrapper tool, *ament_cobra*, that is used to integrate Cobra into the ament-based build and test system.
+That is, when building and executing tests for Space ROS, along with the other source code analysis tools, ament_cobra is invoked to analyze the code using the AUTOSAR C++ 14 ruleset.
```
Comment (nuclearsandwich) [inline source/Related-Projects/Cobra.rst]:
```suggestion
Space ROS adds cobra to the common set of linters used by most of the Space ROS core by including it in its forked version of *ament_lint_common*.
That is, when building and executing tests for Space ROS, along with the other source code analysis tools, ament_cobra is invoked to analyze the code using the AUTOSAR C++ 14 ruleset.
```

The real "secret sauce" step to Cobra being run on Space ROS packages is the inclusion of it in ament_lint_common. Without that cobra and ament_{,cmake}_cobra would be present but not run for packages that don't expressly include it in their test dependencies.

**Review by nuclearsandwich — APPROVED** [review]
I have some small reservations about the introductory section describing cobra being "too general" to include on the Space ROS docs site but it's right on the cusp and not a tool that's generally used in ROS yet so I think it's probably an acceptable overview. Left a few other suggestions inline but it looks good with or without them.

File: source/Related-Projects/Cobra.rst (hunk 1 above)
Comment (mjeronimo) [reply source/Related-Projects/Cobra.rst]: Added.

File: source/Related-Projects/Cobra.rst (hunk 2 above)
Comment (mjeronimo) [reply source/Related-Projects/Cobra.rst]: Yes, I missed that part. Added.

**Comment by mjeronimo** [comment]
I did try to include enough info that readers get the general idea. I'm trying to find the right balance (not going into too much detail, but provide the general idea).

**Pull request merged**